sympy/solvers/solveset.py

```diff
@@ -125,11 +126,62 @@ def _invert_real(f, g_ys, symbol):
                             symbol)
 
     if isinstance(f, Abs):
-        pos = Interval(0, S.Infinity)
-        neg = Interval(S.NegativeInfinity, 0)
-        return _invert_real(f.args[0],
-                    Union(imageset(Lambda(n, n), g_ys).intersect(pos),
-                          imageset(Lambda(n, -n), g_ys).intersect(neg)), symbol)
+        conditions = S.EmptySet
+        condition_interval = Interval(0, oo)
+
+        if f.is_positive:
+            f_n = f
+        else:
+            f_n = f.args[0]
+        g_x, values = _invert_real(f_n, Union(imageset(Lambda(n, n), g_ys),
+                                   imageset(Lambda(n, -n), g_ys)), symbol)
+
+        if isinstance(values, ConditionSet):
```

Comment:
OK, looking more closely I see that much of what you have here is just simplification of ConditionSet. That should probably be in ConditionSet (maybe _eval?).